### PR TITLE
Handle data format and graph compile properly for better GPU performance

### PR DESCRIPTION
**Pull request by haoyuz**
1. When `data_format` is `channels_first`, transpose the input tensors to `NCHW` for better performance on GPU;
2. Compile only the training graph when `skip_eval` flag is `True`. This allows better memory efficiency so that a larger (per-GPU) batch size can be used for training.

### CODE REVIEW

File: official/resnet/keras/keras_imagenet_main.py [hunk 1]
```diff
@@ -28,6 +28,7 @@
 from official.utils.flags import core as flags_core
 from official.utils.logs import logger
 from official.utils.misc import distribution_utils
+from tensorflow.python.keras import backend
```
Comment (guptapriya) [inline official/resnet/keras/keras_imagenet_main.py]: is this needed? we could just do `tf.keras.backend.set_...` ?

File: official/resnet/keras/keras_imagenet_main.py [hunk 2]
```diff
@@ -95,6 +96,8 @@ def run(flags_obj):
     raise ValueError('dtype fp16 is not supported in Keras. Use the default '
                      'value(fp32).')
 
+  backend.set_image_data_format(flags_obj.data_format)
```
Comment (guptapriya) [inline official/resnet/keras/keras_imagenet_main.py]: Can we make all these changes to the corresponding Cifar version too? at least to be consistent. I don't know how much performance benefit there will be in that case.

File: official/resnet/keras/keras_imagenet_main.py (hunk 1 above)
Comment (haoyuz) [reply official/resnet/keras/keras_imagenet_main.py]: Done.

File: official/resnet/keras/keras_imagenet_main.py (hunk 2 above)
Comment (haoyuz) [reply official/resnet/keras/keras_imagenet_main.py]: Done. Added same logic to Cifar model as well.

File: official/resnet/keras/keras_cifar_main.py [hunk 3]
```diff
@@ -160,6 +162,7 @@ def run(flags_obj):
 
   validation_data = eval_input_dataset
   if flags_obj.skip_eval:
+    tf.keras.backend.set_learning_phase(1)
```
Comment (guptapriya) [inline official/resnet/keras/keras_cifar_main.py]: can you also add a comment here explaining why we do this.

**Comment by haoyuz** [comment]
Thanks Priya for the comments! Regarding performance improvement, changing data format to channels_first improves throughput by 6.4%. Just finished convergence test, and the model converges to 76.5% at 90th epoch. With that I will merge this PR.

**Pull request merged**